core/src/main/scala/better/files/UnicodeCharset.scala

```diff
@@ -73,7 +75,9 @@ class BomEncoder(charset: Charset) extends CharsetEncoder(charset, 1, 1) {
     .toArray
   private[this] var isBomWritten = false
 
-  override def encodeLoop(in: CharBuffer, out: ByteBuffer): CoderResult = {
+  override def encodeLoop(_in: CharBuffer | Null, _out: ByteBuffer | Null): CoderResult = {
+    val in  = _in.nn
```

Comment:
**Overridden java arguments are never null** | 2
Argument should never be null, immediately remove nullness of parameter.